### ISSUE TITLE
Perf - Storing preferences causes components re-rendering

### ISSUE THREAD

**Issue by Akarshit** [reactioncommerce/reaction#3477]
### Expected behavior
Only those components should re-render, whose preferences have been changed.
### Actual behavior
All the components that depend on **any** preference are re-rendered when **any** preference is changed.
### Steps to reproduce the behavior
1. Run `reaction`
2. Login as admin
1. Open `Social Settings` from Action menu
1. In the chrome dev tools put a breakpoint [here](https://github.com/reactioncommerce/reaction/blob/master/imports/plugins/core/router/client/browserRouter.js#L158)
1. Toggle the Facebook dropdown.
1. The control reached the breakpoint, which means the whole of UI is going to be tried for re-rendering
**OR** 
1. Add [here](https://github.com/reactioncommerce/reaction/blob/master/client/modules/i18n/startup.js#L155)

```js
Tracker.autorun(function () {
  console.log(Reaction.getUserPreferences("reaction-dashboard", "viewAs"));
});
``` 
1. Add a breakpoint at the above `console.log()` statement
1. Toggle the Facebook dropdown.
1. The control reached the breakpoint, which means all the components depending on any preference will be tried for re-rendering.

**Comment by Akarshit** [comment]
Fixed in #3463

**Comment by zenweasel** [comment]
Closed via #3463
